docs/berkeley-upgrade/migrating-archive-database-to-berkeley.mdx

```diff
@@ -13,225 +13,419 @@ keywords:
 
-- **Stage3:** Remainder migration
+`-t | --target-db`: connection string to the database that will hold the migrated data
 
-Each stage has two migration phases:
+`-b | --blocks-bucket`: name of the precomputed blocks bucket. Precomputed blocks are assumed to be named with format: `{network}-{height}-{state_hash}.json`
 
-- **Phase 1:** Copying data from mainnet database and precomputed blocks using the **berkeley_migration** app.
+`-bs | --blocks-batch-size`: number of precomputed blocks to be fetched at one time from Google Cloud. A larger number, like 1000, can help speed up the migration process.
 
-- **Phase 2:** Populating new berkeley tables using the **replayer app in migration mode**
+`-n | --network`: network name (`devnet` or `mainnet`) when determining precomputed blocks. Precomputed blocks are assumed to be named with format: `{network}-{height}-{state_hash}.json`.
 
-Review these phases and stages before you start the migration:
+`-fc | --fork-config`: fork genesis config file is the new genesis config that is distributed with the new daemon and is published after the fork block is announced
+  
+## Advanced approach
 
-#### Stage 1: Initial migration
+If the simplified berkeley migration script is, for some reason, not suitable for you, it is possible to run the migration using the **berkeley_migration** and **replayer** apps without an interface the script provides.
 
-This stage is the very first one, which requires the initial `berkeley` schema. It is a foundation for the next migration stage because it populates the migrated database and creates an initial checkpoint for further incremental migration.
+### Stage 1: Initial migration
+
+This first stage requires only the initial Berkeley schema, which is the foundation for the next migration stage. This schema populates the migrated database and creates an initial checkpoint for further incremental migration.
 
 - Inputs
-   - Unmigrated mainnet database
-   - Mainnet genesis ledger 
-   - Empty `berkeley` schema 
+   - Unmigrated Devnet/Mainnet database
+   - Devnet/Mainnet genesis ledger 
+   - Empty target Berkeley database with the schema created, but without any content
 
 - Outputs
-   - Migrated `mainnet` database to the `berkeley` database from genesis up to the last canonical block
-   - Replayer checkpoint, which can be used for incremental migration
+   - Migrated Devnet/Mainnet database to the Berkeley format from genesis up to the last canonical block in the original database
+   - Replayer checkpoint that can be used for incremental migration
 
-##### Phase 1: Berkeley migration app run
+#### Phase 1: Berkeley migration app run
 
 ```
-mina-berkeley-migration --batch-size 1000 --config-file genesis_ledgers/mainnet.json
---mainnet-archive-uri {mainnet_connection_string} --migrated-archive-uri {migrated_connection_string} --mainnet-blocks-bucket 
-{bucket name} --network "mainnet"
+mina-berkeley-migration \
+   --batch-size 1000 \
```

Comment:
is it intentional that we use `--batch-size` here? it seems inconsistent because other sections, like [http://localhost:3000/berkeley-upgrade/migrating-archive-database-to-berkeley#stage-1-initial-migration](http://localhost:3000/berkeley-upgrade/migrating-archive-database-to-berkeley#stage-1-initial-migration), use `--blocks-batch-size` 

https://github.com/o1-labs/docs2/pull/881/files#diff-4c398e546900ec09fbecdf745b91f338c0b512a11ca1427e6ec597d1bed21d71R75

Reply:
Those are for different software. For `mina-berkeley-migration-script`, it has the flag `blocks-batch-size` while for `mina-berkeley-migration`, it has the flag `batch-size`